src/django/api/views/facility/facilities_view_set.py

```diff
@@ -604,194 +583,19 @@ def create(self, request):
-                'action': ProcessingAction.GEOCODE,
-                'started_at': geocode_started,
-                'error': True,
-                'message': str(exc),
-                'trace': traceback.format_exc(),
-                'finished_at': str(timezone.now()),
-            })
-            item.save()
-            result['status'] = item.status
-            log.error(f'[API Upload] Geocode Error: {str(exc)}')
-            log.info(f'[API Upload] FacilityListItem Id: {item.id}')
-            log.info(f'[API Upload] Address: {row.address}, '
-                     f'Country Code: {row.country_code}')
-            return Response(result,
-                            status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-
-        if item.status == FacilityListItem.GEOCODED:
-            log.info('[API Upload] Trying to start Match process!')
-            log.info(f'[API Upload] FacilityListItem Id: {item.id}')
-            log.info(f'[API Upload] Source Id: {item.id}')
-            # Handle and produce message to Kafka with source_id data
-            timer = 0
-            timeout = 25
-            fli_temp = None
-            while True:
-                if timer > timeout:
-                    break
-                fli_temp = FacilityListItemTemp.objects.get(
-                    source=source.id
-                )
-                if fli_temp.status == FacilityListItemTemp.GEOCODED:
-                    log.info('[API Upload] Started Match process!')
-                    log.info(f'[API Upload] FacilityListItem Id: {item.id}')
-                    log.info(
-                        f'[API Upload] FacilityListItemTemp Id: {fli_temp.id}'
-                    )
-                    log.info(f'[API Upload] Source Id: {item.id}')
-                    asyncio.run(produce_message_match_process(source.id))
-                    break
-                asyncio.sleep(1)
-                timer = timer + 1
-
-            # Handle results of "match" process from Dedupe Hub
-            result = handle_external_match_process_result(
-                fli_temp.id,
-                result,
-                request,
-                should_create
-            )
-
-        errors_status = [FacilityListItem.ERROR_MATCHING,
-                         FacilityListItem.GEOCODED_NO_RESULTS]
-
-        log.info(f'[API Upload] Result data: {result}')
-        log.info(f'[API Upload] FacilityListItem Id: {item.id}')
-        log.info(f'[API Upload] Source Id: {item.id}')
+        contri_cleaner = ContriCleaner(request.data, SectorCache())
+        processed_data = contri_cleaner.process_data()
+        rows = processed_data.rows
+        row = rows[0]
 
-        if (should_create
-                and result['status'] not in errors_status):
-            return Response(result, status=status.HTTP_201_CREATED)
-        else:
-            return Response(result, status=status.HTTP_200_OK)
+        return CreateFacility.createApi(request, row, source, should_create)
```

Comment:
@killev @mazursasha1990 I'm not sure that **CreateFacility** is a good name for it. As there is not always new facility creation happening, there could be "matched", "pending (potential match)". I suggest name it "ProcessingFacility" or something similar.

Reply:
Good idea

Reply:
You are right. 
Thanks!